blackduck/Authentication.py

```diff
@@ -89,91 +89,76 @@ def authenticate(self):
-        if not self.verify:
+        logger.warning("Authenticating with username/password is not recommended. Consider using the more secure "
+                       "token based authentication instead.")
+        if not self.session.verify:
             requests.packages.urllib3.disable_warnings()
             # Announce this on every auth attempt, as a little incentive to properly configure certs
-            logger.warn("ssl verification disabled, connection insecure. do NOT use verify=False in production!")
-            
-        try:
-            response = self.session.request(
-                method='POST',
-                url='/j_spring_security_check',
-                headers = {
-                    "j_username" : self.username,
-                    "j_password" : self.password
-                },
-            )
-
-            if response.status_code / 100 != 2:
-                self.http_exception_handler(
-                    response=response,
-                    name="authenticate"
-                )
-
-            content = response.json()
-            self.cookie = response.headers.get('Set-Cookie', 'COOKIE_UNSET')
-            self.auth_token = self.cookie[cookie.index('=')+1:cookie.index(';')]
-            self.valid_until = datetime.utcnow() + timedelta(milliseconds=int(content.get('expiresInMilliseconds', 900000)))
-
-        # Do not handle exceptions - just just more details as to possible causes
-        # Thus we do not catch a JsonDecodeError here even though it may occur
-        # - no futher details to give.  
-        except requests.exceptions.ConnectTimeout as connect_timeout:
-            logger.critical(f"could not establish a connection within {self.timeout}s, this may be indicative of proxy misconfiguration")
-            raise connect_timeout
-        except requests.exceptions.ReadTimeout as read_timeout:
-            logger.critical(f"slow or unstable connection, consider increasing timeout (currently set to {self.timeout}s)")
-            raise read_timeout
-        else:
-            logger.info(f"success: auth granted until {self.valid_until} UTC")
\ No newline at end of file
+            logger.warning("ssl verification disabled, connection insecure. do NOT use verify=False in production!")
+
+        credentials = {'j_username': self.username, 'j_password': self.password}
+        response = self.session.post(
+            url="/j_spring_security_check",
+            data=credentials,
+            auth=NoAuth()  # temporarily strip authentication to avoid infinite recursion
+        )
+
+        if response.status_code == 204:  # No Content
+            try:
+                cookie = response.headers['Set-Cookie']
+                self.auth_token = cookie[cookie.index('=')+1:cookie.index(';')]
+                self.valid_until = datetime.utcnow() + timedelta(minutes=120)  # token is good for 2 hours
```

Comment:
Is this definitely the case for all hub instances? if expiresInMilliseconds is still available I think we should prefer that to hard coding.

Reply:
I do not know for sure. Also I could not find any reference to the expiry time in the response.